src/classes/vehicle.py

```diff
@@ -87,73 +90,121 @@ def update_cmd(old_cmd: Command, t: np.array, a: np.array, elapsed_time: float=0
+
     for vehicle in vehicles:
-        leading_vehicle = vehicle.leading_vehicle
-        initial_velocity = vehicle.velocity
-        
-        if leading_vehicle:
-            final_velocity = leading_vehicle.velocity
-            distance = abs(vehicle.route_position - leading_vehicle.route_position) 
+        if vehicle.leading_vehicle:
+            handle_leading_vehicle(vehicle, cur_time)
+        # only accelerate when the vehicle is on a StraightEdge
+        elif vehicle.direction_angle % 90 == 0:
+            handle_no_leading_vehicle(vehicle, cur_time)
 
-            # ALL VEHICLES MUST BE PLACED 10m AWAY FROM EACH OTHER AT THE START
-            
-            # we want to stop the car "Safety Distance" away from leading car
-            safety_distance = 6
-            emergency_distance = 5
-
-            # this prevents division of negative and 0
-            if distance < emergency_distance:
-                vehicle.velocity = 0
-                print("Invalid Implement Command Regarding Emergency Distance")
-                continue
-            elif distance == safety_distance:
-                required_deceleration = 0
-            else:
-                # required deceleration to stop car at fake distance away
-                required_deceleration = (final_velocity**2 - initial_velocity**2) / (2 * (distance - safety_distance))
-            
-            new_t = np.array([cur_time, cur_time + 0.1])
-            new_a = np.array([required_deceleration, leading_vehicle.acceleration])
-            
-            vehicle.command = update_cmd(vehicle.command, new_t, new_a, cur_time)
+        check_traffic_lights(vehicle, cur_time)
+
+
+def handle_leading_vehicle(vehicle, cur_time: float) -> None:
+    """Handle the case where there is a leading vehicle."""
     
-        elif abs(vehicle.velocity - vehicle.default_velocity) > 0.01:
-            acceleration_distance = 10
-            required_deceleration = (vehicle.default_velocity**2 - initial_velocity**2) / (2 * acceleration_distance)
-            
-            new_t = np.array([cur_time, cur_time + 0.1])
-            new_a = np.array([required_deceleration, vehicle.acceleration])   
-            
+    # we want to stop the car "Safety Distance" away from leading car
+    safety_distance = 6
+    emergency_distance = 5
+
+    leading_vehicle = vehicle.leading_vehicle
+    initial_velocity = vehicle.velocity
+    final_velocity = leading_vehicle.velocity
+    distance = abs(vehicle.route_position - leading_vehicle.route_position)
+
+    # this prevents division of negative and 0
+    if distance < emergency_distance:
+        vehicle.velocity = 0
+        print("Invalid command: Distance to leading vehicle is too short")
+        return
+    
+    # required deceleration to stop car at fake distance away
+    required_deceleration = calculate_deceleration(final_velocity, initial_velocity, distance, safety_distance)
+
+    new_t = np.array([cur_time, cur_time + 0.01])
```

Comment:
what is 0.01 for?

Reply:
# since we want to constantly update the vehicle's command, we update it every fps rather than a long range of time 

added this comment